### DISCUSSION TITLE
Locating the Visual Studio SDK for Python build

### DISCUSSION THREAD

**Discussion by rgov** [community/community#25581] [Actions]
<p>I am helping a Python project to switch to GitHub Actions, and for the first time we are able to perform CI builds on Windows.</p>
<p>Our <code>setup.py</code> script uses <a href="https://github.com/pybluez/pybluez/blob/master/setup.py#L39" rel="nofollow noopener">some logic</a> to find Visual Studio depending on the Python version installed.</p>
<p>This detection does not work in the Windows build environment. Is there a better way to do this that will be compatible with GitHub Action’s environment?</p>

**Comment by ethomson** [comment, marked as answer]
<p>We install the Visual Studio build tooling, so that you can use the compilers, linkers, etc.  However, we do not install the Windows SDK.</p>
<p>If you want to install one of the Windows SDKs, the easiest way to do that is to install it with chocolatey.  For example, you can install the <a href="https://chocolatey.org/packages/windows-sdk-10.0" rel="nofollow noopener">https://chocolatey.org/packages/windows-sdk-10.0</a> package and I think that your build script will be able to locate it.</p>

**Comment by rgov** [comment]
<p>Thanks <a class="mention" href="/ethomson">@ethomson</a> . I see in the current GitHub Actions docs that these packages are installed:</p>
<ul>
<li>Microsoft.VisualStudio.Component.VC.v141</li>
<li>Microsoft.VisualStudio.Component.Windows10SDK.16299</li>
</ul>
<p>Do these correspond with the Visual C++ build tools version 14, and also the Windows 10 SDK? If so, is it still necessary to install these with Chocolatey still?</p>
